frontend/src/lib/util.ts

```diff
@@ -107,3 +109,17 @@ export function getCluster(): string | null {
     matchPath<{cluster?: string}>(urlPath, {path: getClusterPrefixedPath()});
   return (!!clusterURLMatch && clusterURLMatch.params.cluster) || null;
 }
+
+export function useErrorState(dependentSetter?: (...args: any) => void) {
+  const [error, setError] = React.useState<ApiError | null>(null);
+
+  React.useEffect(() => {
+    if (!!error && !!dependentSetter) {
+      dependentSetter(null);
+    }
+  },
+  [error, dependentSetter]);
+
+  // Adding "as any" here because it was getting difficult to validate the setter type.
```

Comment:
Why do you need to set the any type on setError isn't it implicit to the useErrorState function?

Reply:
I could validate it by setting the right return type in the function, but that's opening more and more places that need the return types set. Let's get this in now and change it later if a better solution is found.